client/landing/gutenboarding/components/mshots-image/index.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import classnames from 'classnames';
+import { Spinner } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface Props {
+	src: string;
+	alt: string;
+}
+
+const MshotsImage: React.FunctionComponent< Props > = ( { src, alt } ) => {
+	const [ resolvedUrl, setResolvedUrl ] = React.useState< string | undefined >();
+	const [ count, setCount ] = React.useState( 0 );
+	const [ isVisible, setIsVisible ] = React.useState( false );
+
+	const mShotsEndpointUrl = src;
+
+	React.useEffect( () => {
+		async function fetchMshots() {
+			const response = await window.fetch( mShotsEndpointUrl, {
```

Comment:
An alternative, just for your consideration:

```js
const img = new Image();
img.addEventListener('load', (event) => {
 // event
}, false);
img.src = mShotsEndpointUrl;
```

Might help with cors stuff.

Reply:
Oh, that's a nice idea! Unfortunately, I think we might still need to use the fetch approach if we want to hide the fallback `/default` route that renders [the default loading gif](https://s0.wp.com/mshots/v1/default), by handling the 307 redirect case.

It just occurred to me that I haven't looked into the [link prefetch](https://github.com/Automattic/wp-calypso/blob/f392fd8fbdb062b79034b6c2650b5d1bdb687276/client/landing/gutenboarding/available-designs.ts#L62) that we're doing in `prefetchDesignThumbs` — my hunch is that we need to add [the `crossorigin` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content#Cross-origin_fetches), and this might explain why the CORS requests appear to be failing some of the time, but not when I directly reload the Design step. Just a hunch, though! I'll take a quick look early next week and see if it fixes it, if no-one beats me to it!